__init__.py

```diff
@@ -15,95 +15,447 @@
+
+            Args:
+                message_data (dict): a dictionary from the intent message bus
+
+            return:
+                alarm_object (dict): first pass for the alarm data
+        """
+        daytype = message_data.get('daytype', "").replace(" ", "")
+        time = message_data.get('time', "").replace(" ", "")
+        length = message_data.get('length', "")
+        ampm = message_data.get('ampm', "").replace(" ", "")
+        name = None
+        recurring = message_data.get('recurring', False)
+
+        # special case for am and pm as padatious sometimes
+        # match the ampm entity with extra words
+        if ampm != "":
+            if 'pm' in ampm.lower() or 'p.m' in ampm.lower() or \
+               'evening' in ampm.lower():
+                ampm = 'p.m.'
+            if 'am' in ampm.lower() or 'a.m' in ampm.lower() or \
+               'morning' in ampm.lower():
+                ampm = 'a.m.'
+
+        recurring_str = "" if recurring is False else "recurring "
+        if time != "":
+            name = "{}{} {} {}".format(
+                recurring_str, time, ampm, daytype)
+            name = name.strip()
+        elif length is not None:
+            name = "{}".format(length)
+
+        _days = ["monday", "tuesday", "wednesday", "thursday",
+                 "friday", "saturday", "sunday"]
+
+        if 'weekend' in daytype.lower():
+            days = _days[5:7]
+        elif 'weekday' in daytype.lower():
+            days = _days[0:5]
+        elif daytype == "":
+            days = [_days[arrow.now().replace(tzinfo='local').weekday()]]
+        else:
+            days = [_days[i] for i in range(len(_days)) if _days[i] in daytype]
+
+        return {
+            "time": time,
+            "length": length,
+            "daytype": daytype,
+            "ampm": ampm,
+            "name": name,
+            "recurring": recurring,
+            "days": days,
+            "arrow_objects": []
+        }
+
+    def _schedule_alarm_event(self, alarm_name, alarm_time):
+        """ schedules the alarm using event scheduler api from MycroftSkill
+            calls handle_end_timer as a callback when alarm time is up
+
+            Args:
+                alarm_name (str): unique name i.e. 10:30 pm wednesday
+                alarm_time (datetime): datetime object of alarm time
+        """
+        LOG.info("scheduling {} alarm".format(alarm_name))
+        self.schedule_event(self.handle_end_timer, alarm_time,
+                            data=alarm_name, name=alarm_name)
+
+    def schedule_alarm(self, message_data):
+        """ handles scheduling alarm, saving alarm, and
+            speak utterance
+
+            Args:
+                message_data (dict): a dictionary from the intent message bus
+        """
+        alarm_object = self.parse_message_data(message_data)
+
+        if alarm_object['length'] != "":
+            d = dparser.parse(alarm_object["name"])
+            now = datetime.now()
```

Comment:
This gives the local time, but should be working in UTC.  You can use mycroft.util.parser.extract_datetime("") to get UTC now.

Reply:
fixed using 

```python
        user_set_tz = \
            timezone(self.time_zone).localize(datetime.now()).strftime('%Z')
        device_tz = time.tzname
        if user_set_tz in device_tz:
            return arrow.get(d)
        else:
            seconds_to_shift = int(self.location['timezone']['offset']) / -1000
            return arrow.get(d).shift(seconds=seconds_to_shift)
```